File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        
        left = 0
        right = len(height)-1
        
        totalWater = 0
        
        area = 0
        
        
        while left < right:
            smallest = min(height[left], height[right])
    
            area = max(area, smallest*(right-left))
            
            if height[left] >= height[right]:
                right-=1
            else:
                left+=1
            
        return area
            
        

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        
        int left = 0;
        int right = height.size()-1;
        int res = 0;
        
        while (left < right){
            res = max(res, (right-left)*min(height[left], height[right]));
            if (height[left] < height[right])
                left+=1;
            else
                right-=1;
        }
        return res;
    }
};